### PR TITLE
netbsd: move packages into their own modules

### PR DESCRIPTION
**Pull request by ali-abrar**
## Description of changes

Reorganized netbsd packages to follow the pattern established in the freebsd tree.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/bsd/netbsd/default.nix [hunk 1]
```diff
@@ -7,8 +7,6 @@
 }:
```
Comment (Ericson2314) [inline pkgs/os-specific/bsd/netbsd/default.nix:7]: I think some of these imports above are no longer needed?

File: pkgs/os-specific/bsd/netbsd/default.nix [hunk 2]
```diff
@@ -28,9 +26,12 @@ let
...
   otherSplices = generateSplicesForMkScope "netbsd";
-  f = (self: let
-    inherit (self) mkDerivation;
-  in {
+  f = (self: lib.packagesFromDirectoryRecursive {
+    callPackage = self.callPackage;
+    directory = ./pkgs;
+  } // (let inherit (self) mkDerivation; in {
+
+  inherit defaultMakeFlags fetchNetBSD netbsdSetupHook;
```
Comment (Ericson2314) [inline pkgs/os-specific/bsd/netbsd/default.nix:34]: Can we move these 3 from the let block above directly into the inherit? I think nothing else is using them / if things are, we can get them via `self.<name>` instead? we can remove the `let..in` then.

File: pkgs/os-specific/bsd/netbsd/default.nix [hunk 3]
```diff
@@ -28,9 +26,12 @@ let
...
 
   # Why do we have splicing and yet do `nativeBuildInputs = with self; ...`?
   #
```
Comment (Ericson2314) [inline pkgs/os-specific/bsd/netbsd/default.nix:37]: This comment is now partially addressed, and should be combined with the "Overridden arguments avoid cross package-set..." comment in FreeBSD's `default.nix`. We put the combined comment in one place, and reference it in the other place.

File: pkgs/os-specific/bsd/netbsd/default.nix (hunk 2 above)
Comment (Ericson2314) [reply pkgs/os-specific/bsd/netbsd/default.nix:34]: Also now that we basically have all-new and shorter content within this `{....}` we should indent two more spaces to fix the indentation that was broken :).

Comment (Ericson2314) [reply pkgs/os-specific/bsd/netbsd/default.nix:34]: Finally `netbsdSetupHook` can be a package, see `freebsdSetupHook` which is done with a `package.nix` and shell script file,

File: pkgs/os-specific/bsd/netbsd/default.nix (hunk 1 above)
Comment (ali-abrar) [reply pkgs/os-specific/bsd/netbsd/default.nix:5]: removed

File: pkgs/os-specific/bsd/netbsd/default.nix (hunk 2 above)
Comment (ali-abrar) [reply pkgs/os-specific/bsd/netbsd/default.nix]: done

File: pkgs/os-specific/bsd/netbsd/default.nix (hunk 3 above)
Comment (ali-abrar) [reply pkgs/os-specific/bsd/netbsd/default.nix]: updated

**Review by Ericson2314 — APPROVED** [review]

**Comment by alyssais** [comment]
This has broken `pkgsCross.x86_64-freebsd.evdev-proto`:

```
error:
       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:9:12:
            8|
            9|   strict = derivationStrict drvAttrs;
             |            ^
           10|

       … while evaluating derivation 'evdev-proto-x86_64-unknown-freebsd13-6.7'
         whose name attribute is located at /home/qyliss/src/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:331:7

       … while evaluating attribute 'ABI_FILE' of derivation 'evdev-proto-x86_64-unknown-freebsd13-6.7'
         at /home/qyliss/src/nixpkgs/pkgs/os-specific/bsd/freebsd/evdev-proto/default.nix:31:3:
           30|
           31|   ABI_FILE = runCommandCC "abifile" {} "$CC -shared -o $out";
             |   ^
           32|   CLEAN_FETCH_ENV = true;

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: infinite recursion encountered
       at /home/qyliss/src/nixpkgs/pkgs/os-specific/bsd/netbsd/pkgs/mkDerivation.nix:63:25:
           62|   MKRELRO = "no";
           63| } // lib.optionalAttrs (stdenv'.cc.isClang or false) {
             |                         ^
           64|   HAVE_LLVM = lib.versions.major (lib.getVersion stdenv'.cc.cc);
```

**Pull request merged**